plugins/swiotrefactor/src/ad74413r/buffermenu.cpp

```diff
@@ -97,92 +63,39 @@ QString BufferMenu::getInfoMessage()
-{
-	// Y-UNIT-PER-DIV
-	QHBoxLayout *yMinMaxLayout = new QHBoxLayout(settingsWidget);
-	yMinMaxLayout->setMargin(0);
-	yMinMaxLayout->setSpacing(10);
-
-	m_unitPerDivision = new PositionSpinButton(
-		{
-			{unit, 1e0},
-		},
-		"Unit/Div", 10e-6, yMax, false, false, settingsWidget);
-
-	yMinMaxLayout->addWidget(m_unitPerDivision);
-	settingsWidget->layout()->addItem(yMinMaxLayout);
-
-	// Connects
-	connect(m_unitPerDivision, &PositionSpinButton::valueChanged, this, &BufferMenu::setUnitPerDivision);
-	connect(this, &BufferMenu::unitPerDivisionChanged, this, [=, this](double unitPerDiv) {
-		QSignalBlocker block(m_unitPerDivision);
-		m_unitPerDivision->setValue(unitPerDiv);
-	});
-}
-
-void BufferMenu::setAttrValues(QMap<QString, QMap<QString, QStringList>> values) { m_attrValues = values; }
+void BufferMenu::onBroadcastThreshold() {}
 
-QMap<QString, QMap<QString, QStringList>> BufferMenu::getAttrValues() { return m_attrValues; }
+QList<QWidget *> BufferMenu::getWidgetsList() { return m_widgetsList; }
 
-QVector<QBoxLayout *> BufferMenu::getMenuLayers() { return m_menuLayers; }
-
-double BufferMenu::convertFromRaw(int rawValue, QString chnlType)
+CurrentInLoopMenu::CurrentInLoopMenu(QWidget *parent, QString chnlFunction, Connection *conn,
+				     QMap<QString, iio_channel *> chnls)
+	: BufferMenu(parent, chnlFunction, conn, chnls)
 {
-	double value = 0.0;
-	if(m_attrValues.contains(chnlType) && m_attrValues[chnlType].contains("offset") &&
-	   m_attrValues[chnlType].contains("scale")) {
-		double offset = m_attrValues[chnlType]["offset"].first().toDouble();
-		double scale = m_attrValues[chnlType]["scale"].first().toDouble();
-		value = (rawValue + offset) * scale;
+	// dac code - output channel
+	IIOWidget *dacCode = IIOWidgetFactory::buildSingle(IIOWidgetFactory::CMDQAttrData | IIOWidgetFactory::RangeUi,
+							   {.connection = const_cast<Connection *>(m_connection),
+							    .channel = const_cast<iio_channel *>(m_chnls[OUTPUT_CHNL]),
+							    .data = "raw",
+							    .constDataOptions = "[0 1 8191]"},
+							   this);
+	QLayoutItem *item = dacCode->getUiStrategy()->ui()->layout()->itemAt(0);
+	PositionSpinButton *dacSpin = nullptr;
+	if(item) {
+		dacSpin = dynamic_cast<PositionSpinButton *>(item->widget());
 	}
-	return value;
-}
-
-void BufferMenu::onAttrWritten(QMap<QString, QMap<QString, QStringList>> values)
-{
-	m_attrValues = values;
-	Q_EMIT mapUpdated();
-}
-
-CurrentInLoopMenu::CurrentInLoopMenu(QWidget *parent, QString chnlFunction, Connection *conn, iio_channel *chnl)
-	: BufferMenu(parent, chnlFunction, conn, chnl)
-{
-	// dac code
-	QHBoxLayout *dacCodeLayout = new QHBoxLayout();
-	m_dacCodeSpinButton = new PositionSpinButton({{"value", 1E0}}, "DAC Code", 0, 8191, true, false, m_widget);
-	dacCodeLayout->addWidget(m_dacCodeSpinButton);
-	addMenuLayout(dacCodeLayout);
-	// dac label
-	QHBoxLayout *m_dacLabelLayout = new QHBoxLayout();
-	m_dacLabel = new QLabel("mA", m_widget);
```

Comment:
do we still  convert hw units (A) to desired units (mA) for the "current" configured channels?